File: composeApp/src/commonMain/kotlin/org/sesacteamproject/passmate/component/PassmateNavigationRail.kt
package org.sesacteamproject.passmate.component

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.navigationBarsPadding
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.layout.width
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import org.sesacteamproject.passmate.navigation.AppTab
import org.sesacteamproject.passmate.theme.PassmateColors

// Material3 NavigationRail 기본 폭. 오른쪽 1dp 구분선을 포함한 바깥 폭이다
private val RAIL_WIDTH = 80.dp

// 항목 사이 간격 — 하단바의 6:5:5:5:6 비율을 세로로 그대로 쓰면 높이 900dp 창에서 항목이 화면 전체로 흩어진다
private val ITEM_GAP = 8.dp

// 항목 묶음을 바 위 가장자리에서 띄우는 값 — 하단바의 padding(top = 10.dp)와 같은 값을 쓴다
private val ITEM_TOP_PADDING = 10.dp

// 좌측 4탭 레일 (넓은 창) — 하단 탭바를 그대로 세운 형태다. 항목은 PassmateTabItem을 공유하고
// 위에서부터 쌓는다 (스펙 2026-09-16 §3). 표시 여부 판정은 호출부(PassmateNavShell)가 한다
@Composable
fun PassmateNavigationRail(
    selectedTab: AppTab?,
    onSelectTab: (AppTab) -> Unit,
    modifier: Modifier = Modifier
) {
    Row(modifier = modifier.fillMaxHeight().width(RAIL_WIDTH)) {
        Column(
            modifier = Modifier
                .weight(1f)
                .fillMaxHeight()
                .background(PassmateColors.Surface)
                // 레일은 화면 높이를 다 쓰므로 상·하단 시스템 바를 스스로 피한다
                .statusBarsPadding()
                .navigationBarsPadding()
                // 항목은 위에서부터 쌓는다 (사용자 결정 2026-09-16 — 가운데 정렬을 뒤집음)
                .padding(top = ITEM_TOP_PADDING),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(ITEM_GAP)
        ) {
            AppTab.entries.forEach { tab ->
                PassmateTabItem(
                    tab = tab,
                    isSelected = tab == selectedTab,
                    onClick = { onSelectTab(tab) }
                )
            }
        }
        // 하단바의 위쪽 1dp Divider를 90° 돌린 것. CMP 1.5.12에는 VerticalDivider가 없어 Box로 그린다
        Box(
            modifier = Modifier
                .fillMaxHeight()
                .width(1.dp)
                .background(PassmateColors.Border)
        )
    }
}

File: iosApp/iosApp/component/PassmateNavigationRail.swift
import SwiftUI

// 좌측 4탭 레일 (넓은 창) — Compose component/PassmateNavigationRail.kt 미러 (규칙 §14).
// 하단 탭바를 그대로 세운 형태이고 항목은 위에서부터 쌓는다. 표시 여부 판정은 호출부(PassmateNavShell)가 한다.
// 상·하단 시스템 바 회피는 SwiftUI 기본 세이프에어리어가 해준다 (Compose의 statusBarsPadding에 해당)
struct PassmateNavigationRail: View {
    // Material3 NavigationRail 기본 폭. 오른쪽 1pt 구분선을 포함한 바깥 폭이다
    static let width: CGFloat = 80

    // 항목 사이 간격 — 하단바의 6:5:5:5:6 비율을 세로로 그대로 쓰면 높이 900 창에서 항목이 화면 전체로 흩어진다
    private static let itemGap: CGFloat = 8

    // 항목 묶음을 바 위 가장자리에서 띄우는 값 — 하단바의 .padding(.top, 10)과 같은 값을 쓴다
    private static let itemTopPadding: CGFloat = 10

    let selectedTab: AppTab?

    let onSelectTab: (AppTab) -> Void

    var body: some View {
        HStack(spacing: 0) {
            VStack(spacing: Self.itemGap) {
                ForEach(AppTab.allCases, id: \.self) { tab in
                    PassmateTabItemView(
                        tab: tab,
                        isSelected: tab == selectedTab,
                        onTap: { onSelectTab(tab) }
                    )
                }
            }
            // 항목은 위에서부터 쌓는다 — alignment: .top이 없으면 SwiftUI가 가운데로 놓는다
            // (Compose Arrangement.spacedBy(ITEM_GAP) 미러, 사용자 결정 2026-09-16)
            .padding(.top, Self.itemTopPadding)
            .frame(maxWidth: .infinity, maxHeight: .infinity, alignment: .top)
            .background(PassmateColors.surface)
            // 하단바의 위쪽 1pt 구분선을 90° 돌린 것
            Rectangle()
                .fill(PassmateColors.border)
                .frame(width: 1)
        }
        .frame(width: Self.width)
    }
}
